ansible/idr-elasticsearch.yml

```diff
@@ -107,14 +181,34 @@
         node.name: searchengine_elasticsearch_node1
         bootstrap.memory_lock: "true"
         network.host: 0.0.0.0
-        cluster.name: searchengine-cluster
+        cluster.name: "searchengine-cluster"
         cluster.initial_master_nodes: "{{ elasticsearch_nodes | join(',') }}"
         http.host: 0.0.0.0
         #http.port: 9200
         ES_JAVA_OPTS: "-Xms2g -Xmx2g"
         ingest.geoip.downloader.enabled: "false"
+        ########################################
+        es_api_basic_auth_username: "elastic"
+        ELASTIC_PASSWORD: "{{ ELASTIC_PASSWORD }}"
+        es_validate_certs: "no"
+        es_enable_http_ssl: "true"
+        xpack.security.http.ssl.enabled: "true"
+        xpack.security.enabled: "true"
+        xpack.security.authc.realms.file.file1.order: "0"
+        xpack.security.authc.realms.native.native1.order: "1"
+        xpack.security.http.ssl.keystore.path: "/usr/share/elasticsearch/config/certificates/elastic-ca.p12"
+        xpack.security.http.ssl.truststore.password: "{{ ca_password }}"
+        xpack.security.http.ssl.keystore.password: "{{ ca_password }}"
+        xpack.security.transport.ssl.enabled: "true"
+        xpack.security.transport.ssl.verification_mode: "certificate"
+        xpack.security.transport.ssl.keystore.path: "/usr/share/elasticsearch/config/certificates/searchengine_elasticsearch_node1/searchengine_elasticsearch_node1.p12"
+        xpack.security.transport.ssl.truststore.path: "/usr/share/elasticsearch/config/certificates/searchengine_elasticsearch_node1/searchengine_elasticsearch_node1.p12"
+        xpack.security.transport.ssl.keystore.password: "{{ keystore_password }}"
+        xpack.security.transport.ssl.truststore.password: "{{ keystore_password }}"
+      #########################################################################
       networks:
       - name: "searchengine-net"
+        ipv4_address: 10.11.0.2
```

Comment:
What is the requirement for hardcoding this IPv4 address?

Reply:
It is the first searchengine node, so it should have this IP address. The remaining nodes' IPs depended on the number of nodes in the elasticsearch cluster.